graphql_compiler/compiler/blocks.py

```diff
@@ -225,7 +225,8 @@ def __init__(self, direction, edge_name, optional=False, within_optional_scope=F
         Returns:
             new Traverse object
         """
-        super(Traverse, self).__init__(direction, edge_name, optional=optional)
+        super(Traverse, self).__init__(
+            direction, edge_name, optional=optional, within_optional_scope=within_optional_scope)
```

Comment:
Maybe make these args mandatory

Reply:
Yeah I thought about it. Maybe not right now -- I fixed the equality check so the tests will catch problems in those args, and when we move to py3-only, we can prevent them from being passed positionally, so maybe that will be enough.